src/compiler/program.ts

```diff
@@ -2722,6 +2735,14 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
             filesByName.set(path, filesByName.get(oldFile.path));
         });
 
+        if (
+            oldOptions.configFile && oldOptions.configFile === options.configFile ||
+            !oldOptions.configFile && !options.configFile && !optionsHaveChanges(oldOptions, options, optionDeclarations)

```

Comment:
I think you need to check that the program's file set are the same as well, because a new or deleted file can invalidate/introduce some errors. Maybe it's done elsewhere, but it would be good to add a test for that.

Reply:
Early returns above handle that (we wouldn’t be at `StructureIsReused.Completely` if that happened)